### DISCUSSION TITLE
milvus 2.x memory index estimation

### DISCUSSION THREAD

**Discussion by winterfellding** [milvus-io/milvus#26875] [Q&A and General discussion] [answered]
In milvus 1.x, I can estimate the memory usage is about the vector index size, like a million 256 dimension vector data using ivf-sq8 is about 250MB memory need.

But in milvus 2.x, I found that the memory need is a bit higher than the previous version, e.g. if I create a collection with an int64 type id and a 256 dimension vector, and create the ivf-sq8 index, after loading the index to memory, I found that in the metrics the memory usage is increased from 25GB(1.x version) to 37GB, what's the difference between the two versions makes the memory increment, and after some query request, the memory could become a bit higher, is that the goroutine cost?

**Comment by yhmo** [comment]
Milvus v1.x is a pure C++ program.The v2.x is Golang&C++, there are memory copies during pass data between Golang and C++.

**Reply by winterfellding** [reply]
so the query grpc request past the request to golang server, and golang need to load some data from cpp engine, that makes the memory increase when request comes?

another question of mine is how can i estimate how much memory was needed after I define the schema and data size in Milvus 2.x, is there any formula to calculate this memory need.

**Reply by yhmo** [reply, marked as answer]
> so the query grpc request past the request to golang server, and golang need to load some data from cpp engine, that makes the memory increase when request comes?

Yes

> another question of mine is how can i estimate how much memory was needed after I define the schema and data size in Milvus 2.x, is there any formula to calculate this memory need.

The sizing tool is for this purpose

**Reply by winterfellding** [reply]
I get it, thanks a lot for the explanation.
